### PR TITLE
fix: remove now-unusable "legacy" notarization

### PR DESCRIPTION
**Pull request by rotu**
Updated API to reflect deprecated exports.
Updated documentation to match.
Removed code that will no longer work.

### CODE REVIEW

**Comment by rotu** [comment]
note: I took pains to make this backwards-compatible. If you’d rather I remove the obsolete types outright, I could do that instead.

File: src/index.ts [hunk 1]
```diff
@@ -2,58 +2,47 @@ import debug from 'debug';
...
-
-    await notarizeAndWaitForNotaryTool({
-      appPath,
-      ...otherOptions,
-    } as NotaryToolStartOptions);
   }
 
+  await checkSignatures({ appPath });
+
+  d('notarizing using the new notarytool system');
```
Comment (Kilian) [inline src/index.ts:33]: no need for "new" here, maybe just "notarizing with notarytool"?

File: README.md [hunk 2]
```diff
@@ -94,29 +85,8 @@ const password = `@keychain:AC_PASSWORD`;
...
 You can obtain an API key from [Appstore Connect](https://appstoreconnect.apple.com/access/api). Create a key with _App Manager_ access. Note down the Issuer ID and download the `.p8` file. This file is your API key and comes with the name of `AuthKey_<appleApiKeyId>.p8`. This is the string you have to supply when calling `notarize`.
 
-Based on the `ApiKey`, the legacy `altool` will look in the following places for that file:
-
-* `./private_keys`
-* `~/private_keys`
-* `~/.private_keys`
-* `~/.appstoreconnect/private_keys`
-
 `notarytool` will not look for the key, and you must instead provide its path as the `appleApiKey` argument.
```
Comment (Kilian) [inline README.md:88]: since notarytool is the only option, we can replace this sentence with "You must provide its path as the `appleApiKey` argument."

**Comment by Kilian** [comment]
Really appreciate this @rotu!

Had some comments about the docs, someone else has to look at the code bits. I'll leave it to that person whether the error message about legacy should also point to notarytool instead.

**Comment by rotu** [comment]
@MarshallOfSound How do I get code review on this?

**Review by MarshallOfSound — APPROVED** [review]
Seems legit, @rotu we talked about this and based on our policy RE typescript types you can also remove the types in a follow up if you want. This thing doesn't work, removing it can't be considered semver/major in any form. If anything it just fails peoples builds earlier which IMO is better.

**Pull request merged**